.github/workflows/unit-tests-framework.yml

```diff
@@ -20,18 +20,51 @@ concurrency:
   cancel-in-progress: true
 
 jobs:
+  changed-files:
+    if: github.event_name != 'schedule'
+    runs-on: ubuntu-latest
+    outputs:
+      any_changed: ${{ steps.changed-files.outputs.any_changed }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - uses: step-security/changed-files@v46
+        id: changed-files
+        with:
+          base_sha: main
+          files: |
+            !models/**
+            !recipes/**
+            !**.md
+            !.github/**
+            .github/workflows/unit-tests-framework.yml
+
+      - name: Show output
+        run: |
+          echo '${{ toJSON(steps.changed-files.outputs) }}'
+        shell:
+          bash
+
+
   pre-commit:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
-          submodules: "recursive"
       - uses: actions/setup-python@v5
         with:
           python-version: "3.13"
-          cache: "pip"
-      - run: pip install -r requirements-dev.txt
```

Comment:
Are we not doing dev requirements anymore? Maybe the idea is because this happens in the context of unit tests we do not want them there, but in general these get included in build?

Reply:
yeah that's right. we don't need to install them just to run pre-commit 🤷 . pre-commit handles that for us